soroban-sdk/src/storage.rs

```diff
@@ -617,18 +617,39 @@ mod testutils {
             }
             panic!("contract instance for current contract address not found");
         }
+
+        fn get_live_until_ledger(&self) -> u32 {
+            let env = &self.storage.env;
+            env.host()
+                .get_contract_instance_live_until_ledger(env.current_contract_address().to_object())
+                .unwrap()
+        }
     }
 
     impl testutils::storage::Persistent for Persistent {
         fn all(&self) -> Map<Val, Val> {
             all(&self.storage.env, xdr::ContractDataDurability::Persistent)
         }
+
+        fn get_live_until_ledger<K: IntoVal<Env, Val>>(&self, key: &K) -> u32 {
+            let env = &self.storage.env;
+            env.host()
+                .get_contract_data_live_until_ledger(key.into_val(env), StorageType::Persistent)
+                .unwrap()
+        }
     }
 
     impl testutils::storage::Temporary for Temporary {
         fn all(&self) -> Map<Val, Val> {
             all(&self.storage.env, xdr::ContractDataDurability::Temporary)
         }
+
+        fn get_live_until_ledger<K: IntoVal<Env, Val>>(&self, key: &K) -> u32 {
```

Comment:
What does the phrase "live until ledger" mean? The phrase is not particularly clear or intuitive. Is the value returned the ledger it will live till?

Reply:
I documented this as `Gets the ledger until which the temp entry with a given key lives.` . This is the term we've been using for a while in env - TTL describes the *time* to the entry lives (i.e. the number of ledgers), whereas here we're talking specifically about the *sequence* of the ledger until which it lives. But I'm open to suggestions for a more clear name to use in the SDK.

Reply:
`get_ttl` // get number of ledgers it lives
`get_ttl_ledger` // get ledger it lives till

Reply:
`get_live_till_ledger`

Reply:
Hmm, 'till' and 'until' have the exactly same meaning, don't they? (at least according to the online resources I could find with a quick search. 
`get_ttl` looks better, but as I've mentioned TTL is different from the ledger sequence. I actually don't mind returning the TTL instead of the ledger sequence and naming these all `get_ttl` - honestly, I don't have a good intuition as to which one is more convenient to use in the tests. WDYT?

Reply:
I agree, I had the same thought pattern. TTL is instantly recognisable, but yup, it shouldn't return an exact sequence. Given this is for tests I think we have an opportunity to try one and improve it by adding another testutil in a minor version release if it doesn't quite meet the mark.

We could add `get_ttl` today, and if it doesn't satisfy folks use cases, we could add `get_ttl_seq` or `get_archive_ledger_seq` or something else.

@brson thoughts? since you're someone who ran into a need for these utilities.

Reply:
I've changed the names/semantics to TTL. I actually think that makes a bit more sense, given that we have `extend_ttl` functions.

Reply:
I love it. I feel like we're in sync today, I was just about to comment to say about the concept consistency with `extend_ttl` 😆. 👍🏻